examples/util/config.rs

```diff
@@ -64,42 +96,40 @@ impl DbConfig {
     /// Load configuration from a specific file path
     pub fn from_file_path(path: &str) -> io::Result<Self> {
         let content = fs::read_to_string(path)?;
-        let mut endpoint = String::new();
-        let mut database = String::new();
+        let config_file: ConfigFile = toml::from_str(&content)
+            .map_err(|e| io::Error::new(io::ErrorKind::InvalidData, e.to_string()))?;
+
+        let mut config = DbConfig::default();
 
-        for line in content.lines() {
-            let line = line.trim();
-            if line.starts_with("endpoints") {
-                endpoint = line
-                    .split('=')
-                    .nth(1)
-                    .and_then(|s| {
-                        s.trim()
-                            .trim_matches(&['[', ']', '"'][..])
-                            .split(',')
-                            .next()
-                    })
-                    .unwrap_or("127.0.0.1:4001")
-                    .to_string();
-            } else if line.starts_with("dbname") {
-                database = line
-                    .split('=')
-                    .nth(1)
-                    .map(|s| s.trim().trim_matches('"'))
-                    .unwrap_or("public")
-                    .to_string();
+        if let Some(db_config) = config_file.database {
+            if let Some(endpoints) = db_config.endpoints {
+                if let Some(endpoint) = endpoints.first() {
+                    config.endpoint = endpoint.clone();
+                }
+            }
+            if let Some(dbname) = db_config.dbname {
+                config.dbname = dbname;
+            }
+            if let Some(username) = db_config.username {
+                config.username = Some(username.clone());
+            }
+            if let Some(password) = db_config.password {
+                config.password = Some(password.clone());
             }
         }
 
-        Ok(Self {
-            endpoint,
-            dbname: database,
-        })
+        Ok(config)
     }
 
     /// Display current configuration
     pub fn display(&self) {
         println!("Using GreptimeDB endpoint: {}", self.endpoint);
         println!("Using dbname: {}", self.dbname);
+        if let Some(username) = &self.username {
+            println!("Using username: {}", username);
+        }
+        if let Some(password) = &self.password {
+            println!("Using password: {}", password);
```

Comment:
Maybe we should avoid printing passwords.
```suggestion
            println!("Using password: ****");
```